### PR TITLE
Add muticlass F1 support, fix Scorer() test

### PR DESCRIPTION
**Pull request by paroma**
* Add options for multi class F1 score (micro and macro)
* Fix Scorer test that used old label convention

__Test Plan__
* Added to metrics test to cover multi class F1 
* Fix Scorer test case

### CODE REVIEW

File: snorkel/analysis/metrics.py [hunk 1]
```diff
@@ -91,14 +91,21 @@ def _roc_auc_score(golds: np.ndarray, probs: np.ndarray) -> float:
...
+
 # See https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics
 # for details on the definitions and available kwargs for all metrics from scikit-learn
 METRICS = {
     "accuracy": Metric(skmetrics.accuracy_score),
     "coverage": Metric(_coverage_score, ["preds"]),
     "precision": Metric(skmetrics.precision_score),
     "recall": Metric(skmetrics.recall_score),
-    "f1": Metric(skmetrics.f1_score),
+    "f1": Metric(_f1_score, ["golds", "preds"]),
```
Comment (henryre) [inline snorkel/analysis/metrics.py]: if we're hardcoding micro, can we have 3 entries: "f1" that only does skmetrics.f1_score, "f1_micro" that does skmetrics.f1_score(golds, preds, average="micro"), and "f1_macro" that does skmetrics.f1_score(golds, preds, average="macro")?

Comment (paroma) [reply snorkel/analysis/metrics.py]: done

File: snorkel/analysis/metrics.py [hunk 2]
```diff
@@ -91,14 +91,33 @@ def _roc_auc_score(golds: np.ndarray, probs: np.ndarray) -> float:
     return skmetrics.roc_auc_score(golds, probs[:, 0])
 
 
+def _f1_score(golds: np.ndarray, preds: np.ndarray) -> float:
+    if golds.max() <= 1:
+        return skmetrics.f1_score(golds, preds)
+    else:
+        raise ValueError(
+            "f1 not supported for multiclass. Try f1_micro aor f1_macro instead."
```
Comment (henryre) [inline snorkel/analysis/metrics.py]: nit: typo (aor)

File: snorkel/analysis/metrics.py [hunk 3]
```diff
@@ -91,14 +91,33 @@ def _roc_auc_score(golds: np.ndarray, probs: np.ndarray) -> float:
     return skmetrics.roc_auc_score(golds, probs[:, 0])
 
 
+def _f1_score(golds: np.ndarray, preds: np.ndarray) -> float:
+    if golds.max() <= 1:
+        return skmetrics.f1_score(golds, preds)
```
Comment (henryre) [inline snorkel/analysis/metrics.py]: does this fail silently if theres multiple classes?

Comment (paroma) [reply snorkel/analysis/metrics.py]: nope! raises ValueError
